demo/repo/Jenkinsfile

```diff
@@ -17,26 +17,31 @@ parallel(longerTests: {
     runTests(servers, 20)
 })
 
-stage name: 'Staging', concurrency: 1
-node {
-    servers.deploy 'staging'
-}
+milestone ordinal: 1
+lock (resource: 'staging-server', inversePrecedence: true) {
+    stage name: 'Staging'
```

Comment:
Probably you would want this _outside_ the `lock`, so that in the stage view you would see that we are _waiting_ to stage, which I think is how the previous script worked. (The stage view will show delays explicitly if you add `PauseAction` from the `lock` step.)